rest/adminapitest/admin_api_test.go

```diff
@@ -1433,6 +1433,110 @@ func TestResyncStop(t *testing.T) {
 	assert.True(t, syncFnCount < 2000, "Expected syncFnCount < 2000 but syncFnCount=%d", syncFnCount)
 }
 
+// TestCorruptDbConfigHandling:
+//   - Create persistent config rest tester
+//   - Create a db with a non-corrupt config
+//   - Grab the persisted config from the bucket for purposes of changing the bucket name on it (this simulates a
+//     dbconfig becoming corrupt)
+//   - Update the persisted config to change the bucket name to a non-existent bucket
+//   - Assert that the db context and config are removed from server context and that operations on the database GET and
+//     DELETE fail with appropriate error message for user
+//   - Test we are able to update the config to correct the corrupted db config
+//   - asser the db returns to the server context and is removed from corrupt database tracking AND that the bucket name
+//     on the config now matches the rest tester bucket name
+func TestCorruptDbConfigHandling(t *testing.T) {
+	base.LongRunningTest(t)
+	if base.TestsRequireBootstrapConnection() {
+		t.Skip("Walrus/Rosmar does not support persistent config")
+	}
+	numCollections := 1
+	base.RequireNumTestDataStores(t, numCollections)
+
+	rt := rest.NewRestTester(t, &rest.RestTesterConfig{
+		CustomTestBucket: base.GetPersistentTestBucket(t),
+		PersistentConfig: true,
+		MutateStartupConfig: func(config *rest.StartupConfig) {
+			// configure the interval time to pick up new configs from the bucket to every 2 seconds
+			config.Bootstrap.ConfigUpdateFrequency = base.NewConfigDuration(2)
+		},
+	})
+	defer rt.Close()
+
+	ds := rt.CustomTestBucket.GetSingleDataStore()
+	name, ok := base.AsDataStoreName(ds)
+	require.True(t, ok)
+
+	scopesMap := make(map[string]rest.ScopeConfig)
+	collectionMap := make(map[string]rest.CollectionConfig)
+	collectionMap[name.CollectionName()] = rest.CollectionConfig{}
+	scopesMap[name.ScopeName()] = rest.ScopeConfig{
+		Collections: collectionMap,
+	}
+
+	// create db with correct config
+	dbConfig := rest.DbConfig{
+		BucketConfig: rest.BucketConfig{
+			Bucket: base.StringPtr(rt.CustomTestBucket.GetName()),
+		},
+		NumIndexReplicas: base.UintPtr(0),
+		EnableXattrs:     base.BoolPtr(base.TestUseXattrs()),
+		Scopes:           scopesMap,
+	}
+	resp := rt.CreateDatabase("db1", dbConfig)
+	rest.RequireStatus(t, resp, http.StatusCreated)
+
+	// wait for db to come online
+	require.NoError(t, rt.WaitForDBOnline())
+
+	// grab the persisted db config from the bucket
+	databaseConfig := rest.DatabaseConfig{}
+	_, err := rt.ServerContext().BootstrapContext.GetConfig(rt.CustomTestBucket.GetName(), rt.ServerContext().Config.Bootstrap.ConfigGroupID, "db1", &databaseConfig)
+	require.NoError(t, err)
+
+	// update the persisted config to a fake bucket name
+	newBucketName := "fakeBucket"
+	_, err = rt.UpdatePersistedBucketName(&databaseConfig, &newBucketName)
+	require.NoError(t, err)
+
+	// wait for some time for interval to remove the db
+	time.Sleep(2 * time.Second)
```

Comment:
rather than waiting for specific time, can we call a function with a backoff until `GetDbConfig` returns nil? That way if things happen faster we don't have to wait.

Generally a sleep in a test is a bad code smell.